tests/DCPS/DPFactoryQos/run_test.pl

```diff
@@ -12,11 +12,19 @@
 use strict;
 
 my $test = new PerlDDS::TestFramework();
+my $is_rtps_disc = 0;
```

Comment:
tests/DCPS/Messenger is an example that uses PerlDDS::TestFramework (though it's fairly complex since it has a lot of different possible configurations), $is_rtps_disc was left in there but it doesn't do anything.  There are probably better tests to use as an example, though not sure which ones off the top of my head.  You could look at those that have an "rtps_disc" entry in test list and find the ones with minimal number of lines in run_test.pl